src/Components/Components/src/NavigationManager.cs

```diff
@@ -61,7 +61,7 @@ public event EventHandler<NotFoundEventArgs> OnNotFound
     // The URI. Always represented an absolute URI.
     private string? _uri;
     private bool _isInitialized;
-    internal string NotFoundPageRoute { get; set; } = string.Empty;
+    private readonly NotFoundEventArgs _notFoundEventArgs = new();
```

Comment:
We don't want to have one of these, we just want to create a new one every time, since it's "writtable"